services/frontend/src/components/profileActionRibbon/ProfileActionRibbonView.jsx

```diff
@@ -0,0 +1,129 @@
+import { useState } from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+import { useHistory } from "react-router-dom";
+import {
+  EditOutlined,
+  UserAddOutlined,
+  UserDeleteOutlined,
+} from "@ant-design/icons";
+import { Badge, Button } from "antd";
+import PropTypes from "prop-types";
+
+import handleError from "../../functions/handleError";
+import useAxios from "../../utils/useAxios";
+
+import "./ProfileActionRibbonView.less";
+
+const ProfileActionRibbonView = ({
+  children,
+  connectionStatus,
+  loggedInUserId,
+  userId,
+}) => {
+  const [isConnection, setIsConnection] = useState(connectionStatus);
+
+  const history = useHistory();
+  const intl = useIntl();
+  const axios = useAxios();
+  
+  /*
+   * Change Connection
+   *
+   * Adds or removes someone as a connection
+   */
+  const changeConnection = async () => {
+    if (isConnection) {
+      await axios
+        .delete(`api/connections/${userId}`)
+        .catch((error) => handleError(error, "message", history));
+      setIsConnection(false);
+    } else {
+      await axios
+        .post(`api/connections/${userId}`)
+        .catch((error) => handleError(error, "message", history));
+      setIsConnection(true);
+    }
+  };
+
+  /*
+   * Get Action Ribbon Button
+   *
+   * Retrieves the ribbon button for adding/removing a user from one's circle
```

Comment:
please also explain that it displays edit profile when viewing your own profile.